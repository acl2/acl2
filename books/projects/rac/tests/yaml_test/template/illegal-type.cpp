// RAC begin

struct A {
  int a;
};

template <A a>
int foo() {
  return 0;
}
