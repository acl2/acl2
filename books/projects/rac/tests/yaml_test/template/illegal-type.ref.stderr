illegal-type.cpp:7 (10-13): Illegal type `A`, only primitive type (like int or bool) value are allowed as template parameters.
  7 | template <A a>
    |           ^^^
